cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -347,10 +355,29 @@ private void hbaseDelete(List<Delete> deletes) throws IOException {
   protected Scanner scanPersisted(co.cask.cdap.api.dataset.table.Scan scan) throws Exception {
     ScanBuilder hScan = tableUtil.buildScan();
     hScan.addFamily(columnFamily);
-    // todo: should be configurable
-    // NOTE: by default we assume scanner is used in mapreduce job, hence no cache blocks
-    hScan.setCacheBlocks(false);
-    hScan.setCaching(1000);
+
+    // TODO (CDAP-11954): use common utility method to extract these configs
+    if (scan.getProperties().containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
+      hScan.setCacheBlocks(Boolean.valueOf(scan.getProperties().get(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)));
+    } else if (arguments.containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
+      hScan.setCacheBlocks(Boolean.valueOf(arguments.get(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)));
+    } else if (properties.containsKey(CONFIG_HBASE_CLIENT_CACHE_BLOCKS)) {
```

Comment:
Can properties ever have it? Based upon how HBaseTableAdmin#create is implemented, I don't think so.
Or can it have it for inner tables, if the property is defined on the outer dataset?